scipy/stats/tests/test_distributions.py

```diff
@@ -2692,6 +2692,55 @@ def test_minimum_data_zero_gh17801(self):
             _assert_less_or_close_loglike(dist, data, dist.nnlf)
 
 
+class TestPowerNorm:
+
+    # survival function references were computed with mpmath via
+    # from mpmath import mp
+    # x = mp.mpf(x)
+    # c = mp.mpf(x)
+    # float(mp.ncdf(-x)**c)
+
+    @pytest.mark.parametrize("x, c, ref",
+                             [(9, 1, 1.1285884059538405e-19),
+                              (20, 2, 7.582445786569958e-178),
+                              (100, 0.02, 3.330957891903866e-44),
+                              (200, 0.01, 1.3004759092324774e-87)])
+    def test_sf(self, x, c, ref):
+        assert_allclose(stats.powernorm.sf(x, c), ref, rtol=1e-13)
+
+    # inverse survival function references were computed with mpmath via
+    # from mpmath import mp
+    # q = mp.mpf(q)
+    # c = mp.mpf(c)
+    # def isf_mp(q, c):
+    #     arg = q**(1 / c)
+    #     return float(-mp.sqrt(2) * mp.erfinv(2*arg - 1))
+
+    @pytest.mark.parametrize("q, c, ref",
+                             [(1e-5, 20, -0.15690800666514138),
+                              (0.9999, 100, -4.7534143050277855),
+                              (0.9999, 0.02, -2.576676052143387),
+                              (5e-2, 0.02, 17.089518110222244)])
```

Comment:
The tests of `isf` with these value all pass with the default implementation of `_isf`.  We need a test that fails without the changes in this PR, e.g. `q=3e-18`, `c=2`.  `stats.powernorm.isf(3e-18, 2)` returns `inf` with the default implementation. In this PR, we get `5.907940937688639`, and the value computed with mpmath is `5.907940937688638` :+1: .